salt/_states/metalk8s_package_manager.py

```diff
@@ -0,0 +1,36 @@
+"""Custom state for managing yum packages and dependencies"""
+
+from __future__ import absolute_import
+
+import logging
+
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = "metalk8s_package_manager"
+
+
+def __virtualname__():
+    if __grains__['os_family'].lower() == 'redhat':
+        return __virtualname__
+    return (False, "metalk8s_package_manager: no yum based system detected")
+
+
+def install_with_deps(name, version=None, fromrepo=None, **kwargs):
+    """Simple helper to manage package downgrade including dependencies."""
+    if version is None:
+        return __states__["pkg.installed"](
+            name=name, fromrepo=fromrepo, **kwargs
+        )
+
+    dep_list = __salt__['metalk8s_package_manager.list_pkg_deps'](
+        name, version, fromrepo
+    )
+    out = []
+    for pkg, version in dep_list.items():
+        pkg_data = {pkg: version}
+        out.append(pkg_data)
+    return __states__["pkg.installed"](
+        name=name, pkgs=out, fromrepo=fromrepo, **kwargs
```

Comment:
If you want a one liner instead of the for you can do
```suggestion
        name=name, pkgs=[{p: v} for p, v in dep_list.items()], fromrepo=fromrepo, **kwargs
```

Reply:
The list comprehension is a good suggestion, but I would keep it out of the function call to improve the readability:

```python
    pkgs = [{name: version} for name, version in dep_list.items()]
    return __states__["pkg.installed"](
        name=name, pkgs=pkgs, fromrepo=fromrepo, **kwargs
    )
```